Source/WebCore/rendering/EventRegion.cpp

```diff
@@ -110,10 +112,14 @@ bool EventRegionContext::contains(const IntRect& rect) const
 
 #if ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)
 
-void EventRegionContext::uniteInteractionRegions(const Region& region, RenderObject& renderer)
+void EventRegionContext::uniteInteractionRegions(const Region& region, RenderObject& renderer, const PaintInfo* paintInfo)
 {
     if (renderer.page().shouldBuildInteractionRegions()) {
-        if (auto interactionRegion = interactionRegionForRenderedRegion(renderer, region)) {
+        Region clippedRegion = region;
+        if (paintInfo)
+            clippedRegion.intersect(approximateAsRegion(renderer.style().getRoundedBorderFor(paintInfo->rect)));
```

Comment:
`renderer.style().getRoundedBorderFor(paintInfo->rect)` is certainly wrong; this will construct a rounded-cornered rect using the radii in the style for some arbitrary rect which might be a subset of element's rect.

The other question here is whether intersecting with `paintInfo->rect` is valid. When doing the event regions paint pass, do we always start with a layer-sized clipRect, or do we re-use the same, possibly-smaller clip rect that is used for painting, if this is an incremental paint?

Reply:
Also if intersecting with `paintInfo->rect` is correct, I'd expect it to happen out in Renderer code, not down here in EventRegion. And why didn't the event regions paint pass already apply this clip?

Reply:
Maybe that is the real bug: the event regions paint pass is not applying the clip in this case